unleash-yggdrasil/src/lib.rs

```diff
@@ -372,26 +382,41 @@ impl EngineState {
         })
     }
 
-    pub fn resolve(&self, name: &str, context: &Context) -> Option<ResolvedToggle> {
+    pub fn resolve(
+        &self,
+        name: &str,
+        context: &Context,
+        external_values: Option<HashMap<String, bool>>,
+    ) -> Option<ResolvedToggle> {
         self.compiled_state.as_ref().and_then(|state| {
             state.get(name).map(|compiled_toggle| ResolvedToggle {
-                enabled: self.enabled(compiled_toggle, context),
+                enabled: self.enabled(compiled_toggle, context, external_values.clone()),
```

Comment:
Same here, not sure if we can avoid cloning here

Reply:
We can and we should! Good spot, it's borrowed now!